### ISSUE TITLE
Stateful S3-data

### ISSUE THREAD

**Issue by ssalaues** [scality/Zenko#114]
# Stateful Set for S3-data chart

It seems like this chart is setup as a deployment. Is this what we want going forward? It seems like convention would require this to be a stateful set. Would the S3-data service function properly in a stateful set or is the current configuration the only way of achieving a stateful S3-data chart?

**Comment by NicolasT** [comment]
Not necessarily.

The question is: how does the service behave?

If you turn it into a `StatefulSet`, then the PVC spec is part of this object, and the `StatefulSet` controller will create PVCs accordingly for every instance of the set.
As such, when you 'scale up' the number of replicas, this will 'just work'. But maybe the service doesn't support this?

When instead you use a single PVC and a Deployment with a Pod using said PVC, then you *can* only have one instance, or multiple instances but running on the same node, hence sharing and exposing the same storage (assuming `ReadWriteOnce` PVCs).

When this Chart was created, I guess we didn't really know how s3-data behaves or is supposed to behave. I believe the intent was for there to be only one instance, because there's no dispatching going on upfront.

**Comment by ssalaues** [comment]
Yeah I guess we would have to figure out how we would want it to behave vs how it currently behaves then make any appropriate changes to code and charts.
